src/Speech/SpeechClient.php

```diff
@@ -0,0 +1,494 @@
+     * @codingStandardsIgnoreEnd
+     *
+     * @param resource|string|Object $audio The audio to recognize. May be a
+     *        resource, string of bytes, or Google Cloud Storage object.
+     * @param array $options {
+     *     Configuration options.
+     *
+     *     @type int $sampleRate Sample rate in Hertz of the provided audio.
+     *           Valid values are: 8000-48000. 16000 is optimal. For best
+     *           results, set the sampling rate of the audio source to 16000 Hz.
+     *           If that's not possible, use the native sample rate of the audio
+     *           source (instead of re-sampling). Defaults to `8000` with .amr
+     *           files and `16000` with .awb files. If the getID3 library has
+     *           been installed this value will default to the value read from
+     *           the file's headers (if it exists).
+     *     @type string $encoding Encoding of the provided audio. May be one of
+     *           `LINEAR16`, `FLAC`, `MULAW`, `AMR`, `AMR_WB`. Defaults to
+     *           `FLAC` with .flac files, `AMR` with .amr files and `AMR_WB`
+     *           with .awb files.
+     *     @type int $maxAlternatives Maximum number of alternatives to be
+     *           returned. Valid values are 1-30. Defaults to `1`.
+     *     @type string $languageCode The language of the content. BCP-47
+     *           (e.g., en-US, es-ES) language codes are accepted. Defaults to
+     *           en-US (English).
+     *     @type bool $profanityFilter If set to `true`, the server will attempt
+     *           to filter out profanities, replacing all but the initial
+     *           character in each filtered word with asterisks, e.g. \"f***\".
+     *           Defaults to `false`.
+     *     @type array $speechContext Must contain a key `phrases` which is to
+     *           be an array of strings which provide "hints" to the speech
+     *           recognizer to favor specific words and phrases in the results.
+     *           Please see
+     *           [SpeechContext](https://cloud.google.com/speech/reference/rest/v1beta1/RecognitionConfig#SpeechContext)
+     *           for more information.
+     * }
+     * @return Operation
+     * @throws \InvalidArgumentException
+     */
+    public function beginRecognizeOperation($audio, array $options = [])
+    {
+        $response = $this->connection->asyncRecognize(
+            $this->formatRequest($audio, $options)
+        );
+
+        return new Operation(
+            $this->connection,
+            $response['name'],
+            $response
+        );
+    }
+
+    /**
+     * Lazily instantiates an operation. There are no network requests made at
+     * this point. To see the operations that can be performed on an operation
+     * please see {@see Google\Cloud\Speech\Operation}.
+     *
+     * Example:
+     * ```
+     * $operation = $speech->operation('operationName');
```

Comment:
Using `operationName` kind of denotes to me that the name is something a human would create or choose. Maybe just `$operationName` would work here, with a note about where the name comes from.